### PR TITLE
X status warnings for go and python code

### LINKED ISSUE #319: implement x-status constraint
**Is your feature request related to a use case? Please describe.**
x-status constraint is already available in models which will provide the validity of a schema object or a property. Implementation needs to be addressed

**Describe the solution you'd like**
To provide the user the schema object or property is valid, deprecated, obsolete or under review

### PR DESCRIPTION
**Pull request by Vibaswan**
Fixes #319
added x-status support for go and python SDK, added tests as well

### CODE REVIEW

File: do.py [hunk 1]
```diff
@@ -276,7 +276,7 @@ def go_lint():
     )
     run([pkg])
     os.chdir("pkg")
-    run(["golangci-lint run -v"])
+    run(["golangci-lint run -v --disable staticcheck"])
```
Comment (Vibaswan) [inline do.py:279]: had to call disable cause otherwise golangci-lint shows usage of deprecation errors,

File: openapiart/bundler.py [hunk 2]
```diff
@@ -598,6 +598,17 @@ def _required_restriction(self, schema_name, value):
                                 property=required, name=schema_name
                             )
                         )
+                    if "x-status" in value["properties"][required].keys():
```
Comment (Vibaswan) [inline openapiart/bundler.py:601]: added this check as a required property should not have status deprecated, cause it then contradicts with the required nature it-self

File: do.py (hunk 1 above)
Comment (ashutshkumr) [reply do.py:279]: isn't there a different approach ? we cannot disable staticcheck altogether

Comment (Vibaswan) [reply do.py]: included a config file for golangci-lint .golangci.yaml which ignores the static check for deprecation
the code ignored is SA1019 more info in https://staticcheck.io/docs/checks#SA1019

**Review by dipendughosh — APPROVED** [review]
looks good.

**Comment by Vibaswan** [comment]
models branch name: x-status-warnings
successful models build
https://github.com/open-traffic-generator/models/actions/runs/4312828392/jobs/7523872902


snappi branch name: x-status-warnings
successful snappi build
https://github.com/open-traffic-generator/snappi/actions/runs/4312885711

**Comment by ashutshkumr** [comment]
In generated Go code, the deprecation message should be the very first message.

e.g.

Incorrect:
```go
// Space1 returns a int32
// Deprecated: Information TBD
//
// Description TBD
```

Correct:
```go
// Deprecated: Information TBD
// Space1 returns a int32
//
// Description TBD
```

Need to fix following class of x-status as well.
```go
// UpdateConfig is under Review: the whole schema is being reviewed
//
// Object to Test required Parameter
```

**Comment by ashutshkumr** [comment]
I see disparity in naming conventions in generated Go code. Should be `JB` instead of `J_B`. This is sprinkled throughout.

```go
	if obj.obj.Choice.Number() != 0 {
		enumMap := make(map[string]string)
		enumMap["j_b"] = "J_B enum in property Choice is deprecated, use j_a instead"

		for enum, msg := range enumMap {
			if obj.obj.Choice.String() == enum {
				obj.addWarnings(msg)
			}
		}
	}
```

**Review by ashutshkumr — APPROVED** [review]

**Pull request merged**